PhysicsTools/PatUtils/python/tools/muonRecoMitigation.py

```diff
@@ -0,0 +1,91 @@
+        if cleaningScheme in ["bad","all","computeAllApplyBad","computeAllApplyClone"]:
+            setattr(process, 'badGlobalMuonTagger'+typeFix+postfix, badMuModule.clone() )
+            sequence +=getattr(process,"badGlobalMuonTagger"+typeFix+postfix)
+            if cleaningScheme in ["bad","computeAllApplyBad"]:
+                badMuonCollection = 'badGlobalMuonTagger'+typeFix+postfix+':bad'
+        if cleaningScheme in ["clone","duplicated","all","computeAllApplyBad","computeAllApplyClone"]:
+            setattr(process, 'cloneGlobalMuonTagger'+typeFix+postfix, cloneMuModule.clone() )
+            sequence +=getattr(process,"cloneGlobalMuonTagger"+typeFix+postfix)
+            if cleaningScheme in ["clone","duplicated","computeAllApplyClone"]:
+                badMuonCollection = 'cloneGlobalMuonTagger'+typeFix+postfix+':bad'
+        
+        if cleaningScheme=="all":
+            badMuonCollection="badMuons"+postfix
+            badMuonProducer = cms.EDProducer(
+                "CandViewMerger",
+                src = cms.VInputTag(
+                    cms.InputTag('badGlobalMuonTagger'+typeFix+postfix,'bad'),
+                    cms.InputTag('cloneGlobalMuonTagger'+typeFix+postfix,'bad'),
+                    )
+                )
+            setattr(process,badMuonCollection,badMuonProducer)
+            sequence +=getattr(process, badMuonCollection )
+    else:
+        badMuonCollection="badMuons"+postfix
+        badMuonModule = cms.EDFilter("CandViewSelector", 
+                                     src = cms.InputTag(muonCollection), 
+                                     cut = cms.string(selection)
+                                     )
+    
+    # now cleaning ================================
+    cleanedPFCandCollection=cleanCollName+postfix
+    if runOnMiniAOD:
+        cleanedPFCandProducer = cms.EDProducer("CandPtrProjector", 
+                                               src = cms.InputTag(pfCandCollection),
+                                               veto = cms.InputTag(badMuonCollection)
+                                               )
+    else:
+        cleanedPFCandProducer = cms.EDProducer("PFCandPtrProjector", 
+                                               src = cms.InputTag(pfCandCollection),
+                                               veto = cms.InputTag(badMuonCollection)
+                                         )
+        #pfCandTmpPtrs = cms.EDProducer("PFCandidateFwdPtrProducer",
+        #                                      src = cms.InputTag(pfCandCollection)
+        #                                      )
+        #muonTmpPtrs = cms.EDProducer("PFCandidateFwdPtrProducer",
+        #                                      src = cms.InputTag(badMuonCollection)
+        #                                      )
+        #setattr(process,"candPtrTmp"+postfix,pfCandTmpPtrs) 
+        #setattr(process,"muonPtrTmp"+postfix,muonTmpPtrs) 
+        #cleanedPFCandProducer = cms.EDProducer(
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant